### DISCUSSION TITLE
Programatic way to export a signal as CSV?

### DISCUSSION THREAD

**Discussion by ryanbuening** [datalust/seq-tickets#1889] [Help] [answered]
Sorry if this is documented, but I did a brief search and couldn't find an answer.

Is there a way to programatically export a Seq signal as a CSV? Maybe by using the seq cli?

**Comment by nblumhardt** [comment, marked as answer]
Hi Ryan,

Yes, `seqcli` is a good quick way to do this, if you're not dealing with too much data (the default text output format is CSV):

```
> seqcli query -q "select ToIsoString(@Timestamp), @Level, @Message, @Exception from stream limit 3" --signal=signal-2413
"ToIsoString(@Timestamp)","@Level","@Message","@Exception"
"2022-12-08T20:24:35.2296211Z","Information","Order ""order-d34926b6be4a6c7f5e9bfb"" has been shipped; archiving to long-term storage",""
"2022-12-08T20:24:35.8229074Z","Information","Order ""order-b04b5f843f2c51210b881b"" has been shipped; archiving to long-term storage",""
"2022-12-08T20:24:36.3910305Z","Information","Order ""order-e04fffa61b7ca913eca0a1"" has been shipped; archiving to long-term storage",""
```

If you have a lot of data to export or want to do it incrementally, you might be better off writing a small C# app using _Seq.Api_, let me know if you need some more info on this option.

Best regards,
Nick

**Reply by ryanbuening** [reply]
Cool. I was not aware of Seq.Api. Is it possible using the Seq.Api to export a signal as CSV from a specific time period? e.g. The last 90 days?

**Reply by nblumhardt** [reply]
Once you've created a `SeqConnection connection`, [`connection.Data.QueryCsvAsync()` has all of the parameters you need](https://github.com/datalust/seq-api/blob/dev/src/Seq.Api/ResourceGroups/DataResourceGroup.cs#L75) 👍
